awscli/botocore/httpsession.py

```diff
@@ -72,7 +60,13 @@ def create_urllib3_context(ssl_version=None, cert_reqs=None,
         We vendor this function to ensure that the SSL contexts we construct
         always use the std lib SSLContext instead of pyopenssl.
     """
-    context = SSLContext(ssl_version or ssl.PROTOCOL_SSLv23)
+    # PROTOCOL_TLS is deprecated in Python 3.10
+    if not ssl_version or ssl_version == PROTOCOL_TLS:
+        ssl_version = PROTOCOL_TLS_CLIENT
```

Comment:
The only concern here is `PROTOCOL_TLS_CLIENT` doesn't necessarily exist for older versions of urllib3 (<1.25.9). While those are mostly unsupported, we ended up adding a [special import dance](https://github.com/boto/botocore/commit/ddce27a9f4d5e788e6e38cb626b06150aa7cb295) in botocore to avoid failing.

I think this shouldn't be relevant for CLI v2 since everything is distributed, but you might consider the fallback if users are building from source with old urllib3 versions.

Reply:
Thanks. Will check on this since we do technically support `urllib3>=1.25.4`.

Reply:
Added the check here:

https://github.com/aws/aws-cli/pull/7040/files#diff-ac7672d263d603094ead793a19df43bc2a33cdbfe0f5b730489836bd82816950R26

Will confirm next week this is the best approach.